src/core/CompUnit/Repository/Installation.pm6

```diff
@@ -11,13 +11,18 @@ class CompUnit::Repository::Installation does CompUnit::Repository::Locally does
 
     my $verbose := nqp::getenvhash<RAKUDO_LOG_PRECOMP>;
 
+    my sub file-id(Str $name, Str $dist-id) {
+        nqp::sha1($name ~ $dist-id)
+    }
+
     submethod BUILD(:$!prefix, :$!lock, :$!WHICH, :$!next-repo --> Nil) { }
 
     my class InstalledDistribution is Distribution::Hash {
+        has $.dist-id;
         method content($address) {
-            my $entry = $.meta<provides>.values.first: { $_{$address}:exists };
+            my $entry = $.meta<provides>.antipairs.hash{$address};
```

Comment:
Does this actually work? We only upgrade repositories on write. So at this point we can still get the old meta data format.

Reply:
We upgrade repositories during `make install` too.

https://github.com/rakudo/rakudo/blob/68c2c9fea187fcdae33e841ae8a882d722e4804d/tools/build/Makefile-Moar.in#L330